pallets/ctype/src/lib.rs

```diff
@@ -176,11 +186,41 @@ pub mod pallet {
 
 			T::FeeCollector::on_unbalanced(imbalance);
 			log::debug!("Creating CType with hash {:?} and creator {:?}", hash, creator);
-			Ctypes::<T>::insert(hash, creator.clone());
+			Ctypes::<T>::insert(
+				hash,
+				CtypeEntryOf::<T> {
+					creator: creator.clone(),
+					creation_block_number: frame_system::Pallet::<T>::block_number(),
+				},
+			);
 
 			Self::deposit_event(Event::CTypeCreated(creator, hash));
 
 			Ok(())
 		}
+
+		/// Set the creation block number for a given CType, if found.
+		///
+		/// Emits `CTypeUpdated`.
+		#[pallet::weight(<T as pallet::Config>::WeightInfo::set_block_number())]
+		pub fn set_block_number(
+			origin: OriginFor<T>,
+			ctype_hash: CtypeHashOf<T>,
+			block_number: BlockNumberFor<T>,
+		) -> DispatchResult {
+			T::OverarchingOrigin::ensure_origin(origin)?;
+			Ctypes::<T>::try_mutate(ctype_hash, |ctype_entry| -> Result<(), Error<T>> {
```

Comment:
Unnecessary declaration of return function.

```suggestion
			Ctypes::<T>::try_mutate(ctype_hash, |ctype_entry|  {
```

Reply:
https://github.com/KILTprotocol/kilt-node/pull/440/commits/147536f03f84b27c823a8296379d56fb99a2b5f7